jasmine/defs.bzl

```diff
@@ -11,19 +12,54 @@ _jasmine_test = rule(
     toolchains = js_binary_lib.toolchains,
 )
 
-def jasmine_test(jasmine_repository = "jasmine", **kwargs):
+def jasmine_test(
+        name,
+        node_modules,
+        config = None,
+        **kwargs):
+    """jasmine_test rule
+
+    Args:
+        name: A unique name for this target.
+
+        node_modules: Label pointing to the linked node_modules target where jasmine is linked, e.g. `//:node_modules`.
+
+            `jasmine` and `jasmine-reporters` must be linked into the node_modules supplied.
```

Comment:
What if we don't want any reporting?

Reply:
Previously the user didn't need to know that though :(

Reply:
That is not a change from the current behaviour where the attribute is mandatory,

```
    "junit_reporter": attr.label(
        mandatory = True,
        allow_single_file = True,
    ),
```

Reply:
Although, its would be possible to opt-out with an attribute in the macro. I'll add that.

Reply:
Didn't we vendor that package in previously though?

Reply:
We're removing all the vendoring the same as rules_webpack. Its a much better pattern to have users supply their npm deps with these rule sets.

Reply:
Yeah, I know removing it is the goal. I was just mentioning how previously if the user didn't care about reporting they didn't have to do anything, now (at least before adding `generate_junit_xml`) they would have to install a package they don't even want.

Reply:
This is definitely a BREAKING CHANGES for users just like the webpack change but this is pre-1.0 so fair game on a minor release. Long term there will be less maintenance here (and less complexity) since there is no versions.bzl to update with new versions. This also make running tools more similar to how tools are run outside of Bazel which is good and doesn't force users to have to keep versions aligned between their package.json & WORKSPACE file.

Reply:
I don't care about the breakage, I want a good API. Forcing users to install an npm package they don't even know or care about isn't a great API. I think the `generate_junit_xml` solves that fine though 👍